### DISCUSSION TITLE
Dependence Graph 2024-01-22

### DISCUSSION THREAD

**Discussion by ltaoist** [tomlin7/biscuit#218] [Show and tell]
I write a script for inspecting the dependence for the modules internal. Here is the result:

[graph.txt](https://github.com/billyeatcookies/biscuit/files/14113246/graph.txt)

and here is the script:

[search_import_tree.zip](https://github.com/billyeatcookies/biscuit/files/14011371/search_import_tree.zip)

**Comment by ltaoist** [comment]
@billyeatcookies

**Comment by ltaoist** [comment]
![biscuit](https://github.com/billyeatcookies/biscuit/assets/1416332/8a129ef0-1298-4e80-b3e2-dfaf190da13c)

I draw a pictures for some modules/functions/class, @billyeatcookies

**Reply by tomlin7** [reply]
@ltaoist this is super neat and helpful, awesome work ❤️, can you open a pr adding this to the docs?

**Reply by ltaoist** [reply]
@billyeatcookies Hi! I add the graph to the repo and it included in the pr https://github.com/billyeatcookies/biscuit/pull/219/commits/88f5b948c34723278a728709ccd09e843d93c914

It may be too large in a web page visual, here is the pptx files if you want to edit it.
[relations.pptx](https://github.com/billyeatcookies/biscuit/files/14026405/relations.pptx)

Free feel to edit the contents, words, sentences or typos; You are the chief & final editor ❤️.

Good Luck!

**Comment by ltaoist** [comment]
![From_App_Init_To_Run](https://github.com/billyeatcookies/biscuit/assets/1416332/e20a4a10-3c72-4bed-bfba-c0f3a5434f08)

After do an analyse, I make this picture above for explain how the biscuit run up: from App.\_\_init\_\_ to App.run()

And, I noticed that class defined in the code base may divide into some range:

* class defined as component
  * may some sub range further;
  * and some component may also named in manager
* class defined for layout
* class defined as utils
* class defined as editing-wide manager

I think classified that something is _editing-wide_ or something is _feature-wide_ is important. I suggest that do a document about how to distinguish them. And for some new features or issue, how is the stepping to add their into biscuit.

@billyeatcookies
